core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -6823,6 +6824,65 @@ private void checkSemiJoinRuleOnAntiJoin(RelOptRule rule) {
         .checkUnchanged();
   }
 
+  /** Test case for CALCITE-5683 for two level nested decorrelate with standard program
+   * failing during the decorrelation phase. */
+  @Test void testTwoLevelDecorrelate() {
```

Comment:
The test style is different from existing ones, could you change it to be the same style with existing testings?

Reply:
@libenchao Thanks for reviewing the PR. Yeah this test case is different in style because the issue is showing up only with standard program. Let me know if there is a better place(file) to add these test cases.

Reply:
@libenchao Just want to circle back on the review comment.Please let me know if you have any suggestions. Thanks

Reply:
A possible alternative might be putting the tests inside a `*.iq` file (I guess `sub-query.iq` ?). What do you guys think?

Reply:
Since the exception is thrown in `RelDecorrelator`, I would try to enable `withSubQueryRules().withLateDecorrelate(true)` to reproduce the problem. 

`QuidemTest` (*.iq) is also a place we can put the tests, usually I will add tests there if the problem occurs in the 'runtime', or the problem occurs in earlier phase but we also want to verify the result because the plan is not easy to verify correctness.

Reply:
Thanks @HanumathRao , I think it looks good. I have added another comment to see if we can provide more tests for the SqlToRel case.
@libenchao what do you think?